### PR TITLE
BREAKING CHANGE: Lit 2 Upgrade Breaking Changes

### PR DESCRIPTION
**Pull request by dlockhart**
This PR makes the breaking changes required to update to Lit 2 (`lit-html` 2 and `lit-element` 3). The Lit 2 upgrade will be a coordinated effort, so **this PR should not merge at this time**.

Breaking change checklist: 
- [x] Update `lit-html` -> `^2`, `lit-element` -> `^3`, `open-wc/testing` -> `^3` (if used)
- [x] `createRenderRoot` usages -> move from LitElement to ReactiveElement 
- [x] Remove Dependabot Lit rules
- [x] Bump any dependencies to their Lit 2 versions
- [x] Remove any manual lifecycle calls to Lit 1 reactive controllers
- [x] [Backwards-compatible changes merged](https://github.com/BrightspaceUILabs/router/pull/15)

Renamed API's:
- [x] `UpdatingElement` -> `ReactiveElement`
- [x] `@internalProperty` -> `@state` (we don't use TypeScript decorators)
- [x] `static getStyles()` -> `static finalizeStyles(styles)`
- [x] `_getUpdateComplete()` -> `getUpdateComplete()`
- [x] `NodePart` -> `ChildPart`

Testing checklist: 
- [x] CI/ unit tests pass
- [x] Local testing on demo pages
- [ ] Testing component in the LMS

For more information on the Lit 2 upgrade, visit https://lit.dev/docs/releases/upgrade/ or contact team Gaudi.

### CODE REVIEW

**Comment by NicholasHallman** [comment]
To test these breaking changes I can do the following

- update assessment quality dashboard locally to use this version
- point the draco stack lms bsi route to a local bsi hosting the new version of AQD
- check that the router works on the lms

File: example/people/route-loader.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import { html } from 'lit-element';
-import { ifDefined } from 'lit-html/directives/if-defined';
+import { ifDefined } from 'lit-html/directives/if-defined.js';
```
Comment (dlockhart) [inline example/people/route-loader.js:2]: I'd recommend putting this change in a separate non-breaking PR that could merge right away.

**Review by NicholasHallman — APPROVED** [review]

**Comment by dlockhart** [comment]
I think the plan was to wait to merge all these breaking change PRs until we were ready to coordinate the upgrade in ~2 weeks. Probably not the end of the world, but we'll just want to make sure that the things using this don't pick up this new version in the next 2 weeks.

**Comment by NicholasHallman** [comment]
> I think the plan was to wait to merge all these breaking change PRs until we were ready to coordinate the upgrade in ~2 weeks. Probably not the end of the world, but we'll just want to make sure that the things using this don't pick up this new version in the next 2 weeks.

Good to know, fortunatly harmless since this package doesn't use lit in prd, just for the demo. I'll keep what you said in mind for our other projects though

**Pull request merged**